packages/circuit-ui/components/NotificationInline/NotificationInline.tsx

```diff
@@ -155,16 +155,37 @@ const actionButtonStyles = ({ theme }: StyleProps & ButtonProps) =>
 
 const ActionButton = styled(Button)(actionButtonStyles);
 
-const StyledIcon = styled.span(
+const IconWrapper = styled.div(
   ({ theme, variant }: StyleProps & { variant: Variant }) =>
     css`
-      display: block;
+      position: relative;
       align-self: flex-start;
       flex-grow: 0;
       flex-shrink: 0;
       line-height: 0;
       color: ${theme.colors[colorMap[variant]]};
     `,
+  // Adds a black background behind the SVG icon to color just the exclamation mark black.
+  ({ theme, variant }: StyleProps & { variant: Variant }) =>
+    variant === 'notify' &&
+    css`
+      &::before {
+        content: '';
+        display: block;
+        position: absolute;
+        top: 2px;
+        left: 2px;
+        width: calc(100% - 4px);
+        height: calc(100% - 4px);
+        background: ${theme.colors.black};
+        border-radius: ${theme.borderRadius.circle};
+      }
+
+      svg {
+        position: relative;
+        z-index: 1;
+      }
+    `,
 );
```

Comment:
First of we didn’t have to add a new icon, we already have the NotifyCircle in the icons package.
The first approach we tried to change the background color of the svg and cut out the borders, which didn’t work out. The next idea was to use pseudo elements, but it turns out the svg is not supporting pseudo elements.
The last approach was just creating an IconWrapper as a styled div, and adding pseudo element to just add a black background behind the SVG icon to color just the exclamation mark black.

We would like your thoughts and ideas and @robinmetral. It bring also another topic what in case we have to do some of the changes like this more often.

Reply:
I think it makes sense 👍 that's also the implementation I had in mind.

What I'd do is create a `NotifyIcon` component (internal to Circuit) that we can reuse across all components. If we notice that teams also need it we could also export it. WDYT?

Reply:
We thought about that too. However, as Amela pointed out, we might face the same challenge with other icons. Whenever an icon has a color that doesn't meet minimum color contrast requirements, we need a way to turn the shape inside black. The approach here is clunky but works for circular and rectangular icons. What about triangular or oddly shaped icons though? 

Ideally, we'd have a way to change the color of a part of an icon separately from its primary color. This would be possible if, e.g. the exclamation mark of the notify icon was its own path that can be targeted with CSS. The downside here is that the exclamation mark couldn't be transparent anymore. 

Maybe I'm overthinking this?

Reply:
TBH I think that this should be the concern of a team wanting to use custom colors with icons. Just like when using colors anywhere else, they'd need to check contrast. In most cases icons should be the same as the text color so should be 4.5:1 or higher (although icons can actually go down to 3:1, like large text).

I'm not even sure if technically making the `!` black would pass an audit. Users will probably see it more, but the background is still white, really.

So long story short, for now I'd just handle the cases we have inside Circuit go avoid bloating the scope of this thing. But if you're really attached to finding a solution for other potential use cases we could also do some research (starting with, is this really a problem right now?)

What do you think?

Reply:
Sounds good. [KISS](https://www.wikiwand.com/en/KISS_principle).

Reply:
Also just stumbled on this when looking into theming, and it looks like Adobe (Spectrum) found an accessible `notify` color. This would probably be the best moving forward (an inaccessible semantic color is always going to cause issues)

<img width="922" alt="Screenshot 2022-01-26 at 17 25 46" src="https://user-images.githubusercontent.com/35560568/151204206-6c118dc4-35cd-4f60-9408-02860b03a0d8.png">